.eslintrc.js

```diff
@@ -44,8 +55,8 @@ module.exports = {
 				project: ['./tsconfig.eslint.json'],
 			},
 			extends: [
-				'plugin:@typescript-eslint/recommended',
-				'plugin:@typescript-eslint/recommended-requiring-type-checking',
+				'plugin:@typescript-eslint/strict-type-checked',
```

Comment:
We can also add `plugin:@typescript-eslint/stylistic-type-checked` which brings some useful rules but I wasn't sure what rules should be used in the project.